lib/awful/spawn.lua

```diff
@@ -210,6 +212,39 @@ do
     end
 end
 
+local function hash_command(cmd, rules)
+    rules = rules or {}
+    cmd = type(cmd) == "string" and cmd or table.concat(cmd)
+
+    -- Do its best at adding some entropy
+    local concat_rules = nil
+    concat_rules = function (r, max_depth)
+        if max_depth >= 2 then return end
+
+        local keys = gtable.keys(rules)
+
+        for _, k in ipairs(keys) do
+            local v = r[k]
+            local t = type(v)
+
+            if t == "string" or t == "number" then
+                cmd = cmd..k..v
+            elseif t == "tag" then
+                cmd = cmd..k..v.name
+            elseif t == "table" and not t.connect_signal then
+                cmd = cmd .. k
+                concat_rules(v, max_depth + 1)
```

Comment:
Uhm, this recursion does not do anything. `max_depth` (which should just be called `depth` since it is the current depth, not the maximum one) starts out at 1. In the first recursive call, it ends up as `2`, which means that `concat_rules` immediately returns and does nothing.